lib/validators/post_validator.rb

```diff
@@ -25,7 +25,17 @@ def presence(post)
   end
 
   def stripped_length(post)
-    range = post.topic.try(:private_message?) ? SiteSetting.private_message_post_length : SiteSetting.post_length
+    range = if post.topic.try(:private_message?)
+      # private message
+      SiteSetting.private_message_post_length
+    elsif ( post.is_first_post? || (post.topic.present? && post.topic.posts_count == 0 && !post.topic.try(:private_message?)) )
```

Comment:
I don't think you need the `!post.topic.try(:private_message?)` test here since you've already rulled that's  not a PM